rmf_fleet_adapter/include/rmf_fleet_adapter/agv/EasyFullControl.hpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef RMF_FLEET_ADAPTER__AGV__EASYFULLCONTROL_HPP
+#define RMF_FLEET_ADAPTER__AGV__EASYFULLCONTROL_HPP
+
+#include <Eigen/Geometry>
+#include <rmf_fleet_adapter/agv/RobotUpdateHandle.hpp>
+
+namespace rmf_fleet_adapter {
+namespace agv {
+
+//==============================================================================
+class EasyFullControl : public std::enable_shared_from_this<EasyFullControl>
+{
+public:
+
+  using ProcessCompleted = std::function<bool(const std::string& id)>;
```

Comment:
Since a `ProcessCompleted` is returned for a specific request, we do not need to give it a `string id` to get feedback. The internal implementation will store a map.